File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03262/p03262_num3_parsed.cpp
#include <iostream>
#include <vector>
#include <cmath>
#include <algorithm>

int main() {
    int N, X;
    std::cin >> N >> X;
    
    std::vector<int> cities(N);
    for(int i=0; i<N; i++){
        std::cin >> cities[i];
    }
    
    int max_distance = 0;
    for(int i=0; i<N; i++){
        max_distance = std::max(max_distance, std::abs(cities[i] - X));
    }
    
    std::cout << max_distance << std::endl;
    
    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03262/p03262_num3_parsed.py
def max_travel_distance(n, x, cities):
    max_distance = 0
    for city in cities:
        distance = abs(x - city)
        max_distance = max(max_distance, distance)
    return max_distance

n, x = map(int, input().split())
cities = list(map(int, input().split()))

result = max_travel_distance(n, x, cities)
print(result)